include/af/algorithm.h

```diff
@@ -113,6 +214,24 @@ namespace af
     */
     AFAPI array allTrue(const array &in, const int dim = -1);
 
+#if AF_API_VERSION >= 37
+    /**
+       C++ Interface for checking all true values in an array according to a key
+
+       \param[out] keys_out will contain the reduced keys in \p vals along \p dim
+       \param[out] vals_out will contain the reduced and of all values in \p vals along \p dim according to \p keys
+       \param[in] keys is the key array
+       \param[in] vals is the array containing the values to be reduced
+       \param[in] dim The dimension along which the add operation occurs
```

Comment:
"add" should be "all-true" (not sure if there's a better one-word description for this?)